inbm/dispatcher-agent/dispatcher/schedule/sqlite_manager.py

```diff
@@ -19,78 +18,254 @@
+                                        rs.cron_day_month, 
+                                        rs.cron_month, 
+                                        rs.cron_day_week))
+        except (sqlite3.IntegrityError, sqlite3.InternalError, sqlite3.OperationalError) as e:
+            raise DispatcherException(f"Error adding repeated schedule to database: {e}")
+        
+        schedule_id = self._cursor.lastrowid
+        if not schedule_id:
+            raise DispatcherException("No schedule id was added to the repeated_schedule table.")
+        logger.debug(f"Added repeated schedule with id: {str(schedule_id)}, request_id:{rs.request_id}, cron_duration: {rs.cron_duration}, cron_minutes: {rs.cron_minutes}, cron_hours: {rs.cron_hours}, cron_day_month: {rs.cron_day_month}, cron_month: {rs.cron_month}, cron_day_week: {rs.cron_day_week}") # noqa
+        
+        # Add the manifests to the manifest table
+        manifest_ids = self._insert_manifest_to_table(rs.manifests)
+           
+        # Add the schedule_id and manifest_id to the repeated_schedule_manifest table
+        self._insert_repeated_schedule_manifest_tables(schedule_id, manifest_ids)
+        self._conn.commit()
+        
+    def _insert_manifest_to_table(self, manifests: list[str]) -> list[int]:
+        # Add the manifest to the manifest table
+        manifest_ids: list[int] = []
+        
+        if len(manifests) == 0:
+            raise DispatcherException("Error: At least one manifest is required for the schedule.  Manifests list is empty.")
+        
+        for manifest in manifests:
+            sql = ''' INSERT INTO manifest(manifest) VALUES(?); '''
+            try: 
+                self._cursor.execute(sql, (manifest,))
+            except (sqlite3.IntegrityError, sqlite3.InternalError, sqlite3.OperationalError) as e:
+                raise DispatcherException(f"Error inserting manifests into MANIFEST table: {e}")
 
-        # Add the task
-        for manifest in task.manifests:
+            manifest_id = self._cursor.lastrowid
+            if not manifest_id:
+                raise DispatcherException("No manifest id was added to the manifest table.")
+           
+            logger.debug(f"Added manifest with id: {str(manifest_id)}.")
+            manifest_ids.append(manifest_id)
+         
+        if not manifest_ids:
+            raise DispatcherException("No new manifests were added to the manifest table.")
+        
+        return manifest_ids    
+       
+    def _insert_single_schedule_manifest_tables(self, schedule_id: int, manifest_ids: list[int]) -> None:
+        # Add the schedule_id and manifest_id to the join table
+        for manifest_id in manifest_ids:
+            priority = manifest_ids.index(manifest_id)
+            sql = ''' INSERT INTO single_schedule_manifest(priority, schedule_id, manifest_id) VALUES(?,?,?); '''
+            try:
+                self._cursor.execute(sql, (priority, schedule_id, manifest_id))
```

Comment:
**<h2>$\color{#15F4EE}Miscellaneous$</h2>**
**Suggestion:** Close the cursor after its use to free up database resources.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
                self._cursor.execute(sql, (priority, schedule_id, manifest_id))
                self._cursor.close()
```

Reply:
Reading about this, it looks like the garbage collector will close it.  I went ahead and closed it in the destructor before closing the connection.